src/modm/platform/adc/stm32/adc_impl.hpp.in

```diff
@@ -257,3 +257,57 @@ modm::platform::Adc{{ id }}::acknowledgeInterruptFlags(const InterruptFlag_t fla
 {
 	ADC{{ per }}->SR = ~flags.value;
 }
+
+uintptr_t
+modm::platform::Adc{{ id }}::getDataRegisterPointer()
+{
+	return (uintptr_t) &(ADC{{ per }}->DR);
+}
+
+%% if target["family"] not in ["f3"]
+void
+modm::platform::Adc{{ id }}::enableDmaMode()
+{
+	ADC{{ per }}->CR2 |= ADC_CR2_DMA;
+}
+
+void
+modm::platform::Adc{{ id }}::disableDmaMode()
+{
+	ADC{{ per }}->CR2 &= ~ADC_CR2_DMA;
+}
+
+bool
+modm::platform::Adc{{ id }}::getAdcEnabled()
```

Comment:
You sure this doesn't exist already?

Reply:
I couldn't find any other method similar to that, or that at least reads from this bit ion CR2 register.